Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -154,6 +154,19 @@ public extension Sequence {
         return singleElement
     }
 
+    /// SwifterSwift: Remove duplicate elements based on condition.
+    ///
+    ///        [1, 2, 1, 3, 2].withoutDuplicates { $0 } -> [1, 2, 3]
+    ///        [(1, 4), (2, 2), (1, 3), (3, 2), (2, 1)].withoutDuplicates { $0.0 } -> [(1, 4), (2, 2), (3, 2)]
+    ///
+    /// - Parameter condition: conditions to evaluate repeating elements.
+    /// - Returns: Sequence without repeating elements
+    /// - Complexity: O(*n*), where *n* is the length of the sequence.
+    func withoutDuplicates<E: Hashable>(by condition: (Element) -> E ) -> [Element] {
```

Comment:
1. In order to be in line with the other functions, let's make the closure throwable.
2. I don't think that `condition` is the right name for the parameter. You're transforming the element into some hashable value, so `transform` makes more sense to me.
3. `T` is the default generic parameter, unless you want to indicate something more specific. Using `E` confuses it a bit with `Element`, but that's not what `E` represents.
```suggestion
    func withoutDuplicates<T: Hashable>(transform: (Element) throws -> T) rethrows -> [Element] {
```
You'll also need to change the call below to
```Swift
return try filter { !set.insert(try transform($0)).inserted }
```